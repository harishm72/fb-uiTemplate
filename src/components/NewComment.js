import React ,{Component} from 'react';

class NewComment extends Component{

    state = {
        comment : ""
    }
    AddComment = (event) =>{
      this.setState(({comment : event.target.value}))
    }
    SubmitComment = (event) =>{
        event.preventDefault();
        this.props.HandleComment([this.state.comment, Date().toString()], this.props.id)
        this.setState({comment : ""})
    }
    render(){
        
        return(
            <div className="new-comment">
                <form onSubmit={this.SubmitComment}>
                    <input className="comment-input" value={this.state.comment} onChange={this.AddComment} placeholder="type a comment here...."></input>
                </ form>
            </div>
        )
    }
}

export default NewComment;
