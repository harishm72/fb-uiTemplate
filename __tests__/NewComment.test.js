import React from 'react';
import { shallow, mount } from 'enzyme';
import NewComment from '../src/components/NewComment';
import renderer from 'react-test-renderer'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { wrap } from 'module';

Enzyme.configure({ adapter: new Adapter() });

describe('>>>NewComment --- Snapshot', () => {
    it('Rednders the NewComment Component', () => {
        const renderedValue = renderer.create(<NewComment HandleComment={() => null} id ={1}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe("shallow component", () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallow(<NewComment HandleComment={() => null} />)
    })

    it('should have a <form> element', () => {
        expect(wrapper.find('form').length).toBe(1);
    });

    it('<form> element should have an <input/> element', () => {
        expect(wrapper.find('form').childAt(0).type()).toBe('input');
    });
})

describe('add new Comment', () => {
    let wrapper

    it('initially comment box should be empty', () => {
        wrapper = shallow(<NewComment HandleComment={() => null} />)

        let inputBox = wrapper.find('.comment-input')
        expect(inputBox.text()).toBe("");
    })

    it('should change value based on input change', () => {
        wrapper = shallow(<NewComment  HandleComment={() => null} />)
        let inputBox = wrapper.find('.comment-input')

        inputBox.simulate("change", { target: { value: "hello" } })
        expect(wrapper.find('input').at(0).props().value).toEqual("hello");

        inputBox.simulate("change", { target: { value: "hello world" } })
        expect(wrapper.find('input').at(0).props().value).toEqual("hello world");
    })

    it('submit comment and clear input box', () => {
        const fakeEvent = { preventDefault: () => null };
        expect(wrapper.find('.comment-input').length).toBe(1);
        
        wrapper.find('form').at(0).simulate('submit', fakeEvent);
        expect(wrapper.find('.comment-input').text()).toBe("");
    });

})