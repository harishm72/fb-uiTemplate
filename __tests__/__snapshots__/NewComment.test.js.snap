// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>NewComment --- Snapshot Rednders the NewComment Component 1`] = `
<div
  className="new-comment"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="comment-input"
      onChange={[Function]}
      placeholder="type a comment here...."
      value=""
    />
  </form>
</div>
`;
